mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/TaskExecutorState.java

```diff
@@ -100,10 +100,10 @@ boolean onRegistration(TaskExecutorRegistration registration) {
             this.registration = registration;
             this.gateway =
                 rpcService.connect(registration.getTaskExecutorAddress(), TaskExecutorGateway.class)
-                    .whenComplete((gateway, throwable) -> {
-                        if (throwable != null) {
-                            log.error("Failed to connect to the gateway", throwable);
-                        }
+                    .exceptionally((throwable) -> {
```

Comment:
curious, why not keeping `whenComplete`?

Reply:
whenComplete throws the exception to the executing thread.

Reply:
and here we don't need that un-controlled behavior.